imgui/src/input/keyboard.rs

```diff
@@ -94,31 +94,35 @@ impl<'ui> Ui<'ui> {
     /// Returns the key index of the given key identifier.
     ///
     /// Equivalent to indexing the Io struct `key_map` field: `ui.io().key_map[key]`
-    pub fn key_index(&self, key: Key) -> u32 {
-        unsafe { sys::igGetKeyIndex(key as i32) as u32 }
+    fn key_index(&self, key: Key) -> i32 {
+        unsafe { sys::igGetKeyIndex(key as i32) }
```

Comment:
I do wonder if we should be doing `ui.io().key_map[key]` instead, which would allow for the compiler to optimize better.

Oh well, there are a lot of places like that in here, and this is by far not the worst offender

Reply:
hm would probably be best to look at cargo asm or sth. like that, not sure where the key_map is populated (seems its just internal to imgui and then accessed by casting a pointer to the Io), so its not compile time known so it may not be that much difference, just two ways of accessing the array...

Reply:
The compiler will isn't ever able to inline calls to external functions, nor is it allowed to remove them, or reorder accesses around them unless it can prove those accesses are unrelated (which it generally can't).

So, it's good to avoid external calls when you can.